ingestors/nuget.go

```diff
@@ -0,0 +1,132 @@
+package ingestors
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/librariesio/depper/data"
+)
+
+const nugetSchedule = "*/5 * * * *"
+const nugetIndexUrl = "https://api.nuget.org/v3/catalog0/index.json"
+
+type nugetIndex struct {
+	IndexId string `json:"@id"`
+	Pages   []struct {
+		Url             string `json:"@id"`
+		CommitTimeStamp string `json:"commitTimeStamp"`
+		CommitTime      time.Time
+	} `json:"items"`
+}
+
+type nugetPage struct {
+	PageId   string `json:"@id"`
+	Packages []struct {
+		Url             string `json:"@id"`
+		CommitTimeStamp string `json:"commitTimeStamp"`
+		CommitTime      time.Time
+		Name            string `json:"nuget:id"`
+		Version         string `json:"nuget:version"`
+	} `json:"items"`
+}
+
+type Nuget struct {
+	LatestRun time.Time
+}
+
+func NewNuget() *Nuget {
+	return &Nuget{}
+}
+
+func (ingestor *Nuget) Schedule() string {
+	return nugetSchedule
+}
+
+func (ingestor *Nuget) Ingest() []data.PackageVersion {
+	// Until we save LatestRun state, go back to a 4x multiple of the frequency: 20 min.
+	if ingestor.LatestRun.IsZero() {
+		ingestor.LatestRun = time.Now().Add(-20 * time.Minute)
+	}
+	packages := ingestor.ingestURL(nugetIndexUrl)
+	ingestor.LatestRun = time.Now()
+	return packages
+}
+
+func (ingestor *Nuget) ingestURL(url string) []data.PackageVersion {
+	var results []data.PackageVersion
+
+	results, err := ingestor.getIndex(url)
+	if err != nil {
+		log.WithFields(log.Fields{"ingestor": "nuget", "error": err}).Error()
+	}
+
+	return results
+}
+
+func (ingestor *Nuget) getIndex(url string) ([]data.PackageVersion, error) {
+	var results []data.PackageVersion
+
+	response, err := http.Get(url)
+	if err != nil {
+		return results, err
+	}
+	defer response.Body.Close()
+
+	var index nugetIndex
+	body, _ := ioutil.ReadAll(response.Body)
+	err = json.Unmarshal(body, &index)
+	if err != nil {
+		return results, err
+	}
+
+	for _, page := range index.Pages {
+		page.CommitTime, _ = time.Parse(time.RFC3339, page.CommitTimeStamp)
+		if page.CommitTime.After(ingestor.LatestRun) {
```

Comment:
This is interesting and works. I guess I don't know why we need it if we're handling this on the publishing side of things. So far the pattern has been 

ingest: Spit out whatever you want, including way overshooting of duplicate data
publish: Only publish what we haven't seen 

Does that break down in any way here?